core/src/main/kotlin/org/evomaster/core/output/service/HttpWsTestCaseWriter.kt

```diff
@@ -594,8 +594,8 @@ abstract class HttpWsTestCaseWriter : WebTestCaseWriter() {
             val k = when{
                 //TODO should do check for when there are spaces in the field name
                 format.isJavaOrKotlin() -> if(fieldPath.isEmpty()) "" else "'$fieldPath'."
-                format.isJavaScript() -> if(fieldPath.isEmpty()) "" else ".$fieldPath"
-                format.isCsharp()  -> if(fieldPath.isEmpty()) "" else ".$fieldPath"
+                format.isJavaScript() -> if(fieldPath.isEmpty()) "" else "${if(fieldPath.startsWith("["))"" else "."}$fieldPath"
```

Comment:
what is this for? can you add comment to clarify (eg with example)?

Reply:
I added a test `testTestWithObjectAssertion` in `TestCaseWriterTest`. 
for example,  `expect(Object.keys(res_0.body[0]).length).toBe(0);` (see line 1294) would be `expect(Object.keys(res_0.body**.**[0]).length).toBe(0);` with the previous implementation.